spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/core/convert/TestEntities.java

```diff
@@ -105,9 +105,52 @@
 		String value;
 	}
 
+	static class SimpleConstructorTester {
+		@PrimaryKey
+		final String id;
+
+		SimpleConstructorTester(String id) {
+			this.id = id;
+		}
+	}
 
 	static class TestEntityWithListWithZeroTypeArgs {
 		@PrimaryKey
 		List zeroArgsListOfObjects;
 	}
+
```

Comment:
Are constructors where the type of the arguments don't directly match the type of the properties supported?

for example 
`this.stringLength = givenString.length();` 
instead of `this.string = givenString;`

Reply:
That's a really good test case, I think it should work, but let me add it.

Reply:
I tried it out and it is not supported. Mainly because I'm using `PersistentEntityParameterValueProvider` (similar to MongoDb) that assumes that constructor parameters are mapped to the properties of the entity. We can file a separate issue for making this work.